teos/src/main.rs

```diff
@@ -111,24 +111,19 @@ async fn main() {
 
     if is_default {
         log::info!("Loading default configuration")
-    // log the non-default options that the daemon is using
     } else {
-        let default_conf = Config::default();
-        if conf.btc_rpc_port != default_conf.btc_rpc_port {
-            log::info!("Using non-default Bitcoin RPC port: {}", conf.btc_network);
-        }
-        if conf.btc_rpc_user != default_conf.btc_rpc_user {
-            log::info!("Using non-default Bitcoin RPC user: {}", conf.btc_rpc_user);
-        }
-        if conf.btc_rpc_password != default_conf.btc_rpc_password {
-            log::info!("Using non-default Bitcoin RPC password: {}", conf.btc_rpc_password);
-        }
-        if conf.btc_network != default_conf.btc_network {
-            log::info!("Using non-default Bitcoin network: {}", conf.btc_network);
-        }
-        if conf.btc_rpc_connect != default_conf.btc_rpc_connect {
-            log::info!("Using non-default Bitcoin RPC connect address: {}", conf.btc_rpc_connect);
-        }
+        log::info!("Loading configuration from file")
+    }
+
+    // logging the non-default options
+    let non_default_options = conf.non_default_options();
```

Comment:
You are only using this in the following for loop, there is no need to capture it

Reply:
Just call it when declaring the loop

Reply:
> Just call it when declaring the loop

done please check the latest commit once.